dap-mode.el

```diff
@@ -1590,7 +1590,8 @@ before starting the debug process."
                    :wait-for-port :type :request :port
                    :startup-function :environment-variables :hostName host) launch-args)
           (session-name (dap--calculate-unique-name name (dap--get-sessions)))
-          (default-directory (or cwd default-directory))
+          (default-directory (or (and cwd (expand-file-name cwd))
```

Comment:
default-directory should be expanded too.

Reply:
Why that? From its doc: "it should be an absolute directory name". Also, against what would `default-directory` be expanded, given that we omit it here? The doc of `expand-file-name` says that its argument is expanded against `default-directory` unless the second argument is specified.

All this would achieve is that "." and ".." paths get eliminated. This way, we don't change `default-directory` at all unless `:cwd` is specified.

Reply:
Your fix for `dap-chrome` would still behave exactly the same way, because it specifies a `default-directory` `:cwd` that is then needlessly expanded against `default-directory`.

Reply:
For the same reason we expand default-directory in dap-chrome.el - default directory could be `~/foo/bar`

Reply:
Also, the PR is introducing regression - we will send `~/foo/bar` to the adapter since :cwd won't be expanded.

Reply:
To elaborate a bit more, the proper fix is to expand default-directory as described in the root comment and then set cwd in the template.

Reply:
This will still work though:
```emacs-lisp
(let ((default-directory "~/Downloads")) (shell-command "pwd")) ;; => `message`s "/home/nikita/Downloads" on my system"
```

I think there are five options here:
- `expand-file-name` in `dap-chrome`
- always send an expanded :cwd, even if unspecified
- send an expanded :cwd only if specified
- Either of the above, but also expand `default-directory` (against itself)
:cwd would be expanded in all cases. I ended up going with the last one, always sending an expanded :cwd and always expanding `default-directory`.

Reply:
The cwd should be set in the template otherwise commands like restart session won't work in the expected manner when cwd is relative.

Reply:
Why does `dap-start-debugging-noexpand` mutate LAUNCH-ARGS with `plist-put :name`?

Reply:
What is the problem caused by that?

Reply:
> Why does `dap-start-debugging-noexpand` mutate LAUNCH-ARGS with `plist-put :name`?

Modifying the template should be fine and expected since we are copying it.